lib/src/option.dart

```diff
@@ -70,6 +75,11 @@ abstract class Option<T> extends HKT<_OptionHKT, T>
         Filterable<_OptionHKT, T> {
   const Option();
 
+  /// Initialize a **Do Notation** chain.
+  // ignore: non_constant_identifier_names
+  factory Option.Do(DoFunctionOption<T> f) =>
+      Option.tryCatch(() => f(_doAdapter));
```

Comment:
This will swallow any errors thrown outside of the adaptor. I think we should check if the error is a `None` and only handle those.

In this case, the `"fail"` error should still be thrown:
```dart
Option.Do(($) {
  throw "fail";
});
```